replaceActions.js

```diff
@@ -1,93 +1,31 @@
 module.exports = function (input, action, comment) {
-  let runs = input.contents.items.filter((n) => n.key == "runs");
-  if (runs.length) {
-    return replaceInComposite(input, action, comment);
-  }
-  return replaceInWorkflow(input, action, comment);
-};
-
-function replaceInReusable(input, action) {}
-
-function replaceInComposite(input, action, comment) {
-  let actionString = `${action.owner}/${action.repo}`;
-  if (action.path) {
-    actionString += `/${action.path}`;
-  }
-
-  const replacement = `${actionString}@${action.newVersion}`;
+  const { currentVersion, owner, newVersion, path, repo } = action;
 
-  actionString += `@${action.currentVersion}`;
+  const actionId = `${owner}/${repo}${path ? `/${path}` : ""}`;
+  const newComment = generateComment(action, comment);
 
-  const runs = input.contents.items.filter((n) => n.key == "runs")[0].value
-    .items;
-  const steps = runs.filter((n) => n.key == "steps")[0].value.items;
+  const regexpUpdate = new RegExp(
+    `uses(\\s*):(\\s+)${actionId}@${currentVersion}(\\s*)#[^\\n]*`,
```

Comment:
I was working on something related recently and started using [escape-string-regexp](https://www.npmjs.com/package/escape-string-regexp) (to sanitize the dynamic parts of the expression, `actionId` and `currentVersion` in this case), it might make sense to use it here too - what do you think?

Reply:
Worth a try

Reply:
Added in 2947769098f24a7f245d7cdbcf7bacdb257fa2ba. I had to use v4 of the library because v5 (latest) is only available as ESM which is not compatible with this project (relatedly, see https://github.com/mheap/pin-github-action/pull/165#issuecomment-2356087966).